leptonai/queue.py

```diff
@@ -0,0 +1,203 @@
+        """
+        conn = conn if conn else workspace_api.current_connection()
+        res = queue_api.list_queue(conn)
+        if not res.ok:
+            raise RuntimeError(
+                f"Failed to list queues. Error: {res.status_code} {res.content}."
+            )
+        else:
+            return [s["name"] for s in res.json()]
+
+    def __init__(
+        self,
+        name: str,
+        create_if_not_exists: bool = False,
+        error_if_exists: bool = False,
+        conn: Optional[Connection] = None,
+    ):
+        """
+        Initializes a queue.
+
+        :param str name: the name of the queue
+        :param bool create_if_not_exists: if True, create the queue if it does not exist
+        :param bool error_if_exists: if True, raise an error if the queue already exists
+        :param Connection conn: the connection to use. If None, use the default workspace connection.
+        """
+        self._conn = conn if conn else workspace_api.current_connection()
+        res = queue_api.list_queue(self._conn)
+        if not res.ok:
+            raise RuntimeError(
+                "Failed to access queue server. Error:"
+                f" {res.status_code} {res.content}."
+            )
+        existing_queues = [s["name"] for s in res.json()]
+        if name in existing_queues:
+            if error_if_exists:
+                raise ValueError(
+                    f"queue {name} already exists, and you have specified"
+                    " error_if_exists=True."
+                )
+            else:
+                self._queue = name
+        else:
+            if create_if_not_exists:
+                res = queue_api.create_queue(self._conn, name)
+                if not res.ok:
+                    raise RuntimeError(
+                        f"Failed to create queue {name}. Error:"
+                        f" {res.status_code} {res.content}."
+                    )
+                else:
+                    self._queue = name
+
+                # wait for the queue to be ready
+                start = time.time()
+                while (
+                    not self.ready()
+                    and time.time() - start < _lepton_readiness_timeout_
+                ):
+                    logger.trace(f"Queue {name} is not ready yet. Waiting...")
+                    time.sleep(_lepton_readiness_wait_time_)
+                if not self.ready():
+                    raise RuntimeError(
+                        f"Queue {name} is not ready after"
+                        f" {time.time() - start} seconds."
+                    )
+            else:
+                raise ValueError(
+                    f"Queue {name} does not exist, and you have specified"
+                    " create_if_not_exists=False."
+                )
+
+    def ready(self) -> bool:
+        existing_queues = Queue.list_queue(self._conn)
+        return self._queue in existing_queues
+
+    async def async_wait(self):
```

Comment:
Do we need this method? I think a queue instance is always ready to be use after creation (we wait for ready in `__init__`)

Reply:
my fault, I was refactoring it from the kv impl and did not include the wait argument in `__init__`